packages/quantic/cypress/e2e/default-2/recommendation-list/recommendation-list.cypress.ts

```diff
@@ -0,0 +1,122 @@
+    interceptSearchIndefinitely();
+    cy.visit(pageUrl);
+    configure();
+  }
+
+  function setRecommendationsAlias() {
+    cy.wait(getQueryAlias('search')).then((interception) => {
+      const recommendations = extractResults(interception.response);
+      cy.wrap(recommendations).as(recommendationsAlias.substring(1));
+    });
+  }
+
+  describe('while loading the recommendations', () => {
+    it('should properly render the placeholders', () => {
+      setupWithPauseBeforeRecommendationFetch();
+
+      Expect.displayPlaceholders(defaultNumberOfRecommendations);
+      Expect.displayRecommendations(false);
+    });
+  });
+
+  describe('with default options', () => {
+    it('should properly display the recommendations', () => {
+      visitPage({
+        label,
+      });
+      setRecommendationsAlias();
+
+      scope('when loading the page', () => {
+        Expect.events.receivedEvent(true, registerRecommendationTemplatesEvent);
+        Expect.displayLabel(label, defaultHeadingLevel);
+        Expect.displayRecommendations(true);
+        Expect.recommendationsEqual(recommendationsAlias);
+        Expect.correctFieldsIncluded(defaultFieldsToInclude.split(','));
+        Expect.correctRecommendationId(defaultRecommendationId);
+        Expect.correctNumberOfRecommendations(defaultNumberOfRecommendations);
+        Expect.correctNumberOfRecommendationsPerRow(
+          defaultNumberOfRecommendationsPerRow
+        );
+        Actions.clickRecommendationLink(0);
+        Expect.logRecommendationOpen();
+      });
+    });
+  });
+
+  describe('with custom options', () => {
+    const customFieldsToInclude =
+      'source,language,sfcasestatus,sfcreatedbyname';
+    const customNumberOfRecommendations = 3;
+    const customNumberOfRecommendationsPerRow = 1;
+    const customRecommendationId = 'Custom Recommendation Id';
+    const customHeadingLevel = 3;
+
+    it('should properly display the recommendations', () => {
+      visitPage({
+        fieldsToInclude: customFieldsToInclude,
+        numberOfRecommendations: customNumberOfRecommendations,
+        numberOfRecommendationsPerRow: customNumberOfRecommendationsPerRow,
+        recommendation: customRecommendationId,
+        headingLevel: customHeadingLevel,
+        label,
+      });
+      setRecommendationsAlias();
+
+      scope('when loading the page', () => {
+        Expect.events.receivedEvent(true, registerRecommendationTemplatesEvent);
+        Expect.displayLabel(label, customHeadingLevel);
+        Expect.displayRecommendations(true);
+        Expect.recommendationsEqual(recommendationsAlias);
+        Expect.correctFieldsIncluded(customFieldsToInclude.split(','));
+        Expect.correctRecommendationId(customRecommendationId);
+        Expect.correctNumberOfRecommendations(customNumberOfRecommendations);
+        Expect.correctNumberOfRecommendationsPerRow(
+          customNumberOfRecommendationsPerRow
+        );
```

Comment:
should we not add the 

        Actions.clickRecommendationLink(0);
        Expect.logRecommendationOpen();

here? Dont we want to check that it logRecommendationOpen with customFields also?

Reply:
yeah we can add this here as well.
We already checked that the results displayed are indeed the right results to display so I don't think checking the fields sent in the analytics is necessary.